src/docGenerator.ts

```diff
@@ -0,0 +1,132 @@
+          : fromEslintCategoryToCategory(eslintCategory)
+        const parameters =
+          meta && meta.schema
+            ? this.fromEslintSchemaToParameters(meta.schema)
+            : undefined
+        return new PatternsEntry(
+          patternIdToCodacy(patternId),
+          level,
+          category,
+          parameters && parameters.length > 0 ? parameters : undefined
+        )
+      }
+    )
+    return new Patterns(toolName, toolVersion, entries)
+  }
+
+  generateDescription(): DescriptionEntry[] {
+    const descriptionEntries = Array.from(this.rules.entries()).map(
+      ([patternId, ruleModule]) => {
+        const eslintDescription =
+          ruleModule && ruleModule.meta && ruleModule.meta.docs
+            ? ruleModule.meta.docs.description
+            : undefined
+        const description = eslintDescription
+          ? capitalize(eslintDescription)
+          : ""
+        const title = patternTitle(patternId)
+        const timeToFix = 5
+        return new DescriptionEntry(
+          patternIdToCodacy(patternId),
+          title,
+          description,
+          timeToFix
+        )
+      }
+    )
+    return descriptionEntries.filter(x => x !== null) as DescriptionEntry[]
+  }
+
+  private fromEslintSchemaToParameters(
+    schema: JSONSchema4 | JSONSchema4[]
+  ): PatternsParameter[] {
+    const anyOfToArray = (schema: JSONSchema4) => {
+      return schema.anyOf ? schema.anyOf : [schema]
+    }
+
+    const flattenSchema = <JSONSchema4[]>(
+      (<unknown>flatMapDeep(schema, anyOfToArray))
+    )
+
+    if (Array.isArray(flattenSchema)) {
+      const objects = flattenSchema.filter(value => value && value.properties)
+      return fromSchemaArray(objects)
+    }
+    return []
+  }
+
+  private patternIdsWithoutPrefix(prefix: string): Array<string> {
+    const longPrefix = prefix + "/"
+    const patternIds = Array.from(this.rules.entries()).map(e => e[0])
+    const filteredPatternIds = patternIds.filter(patternId =>
+      patternId.startsWith(longPrefix)
+    )
+    return filteredPatternIds.map(patternId =>
+      patternId.substring(longPrefix.length)
+    )
+  }
+
+  private eslintPatternIds(): Array<string> {
+    return Array.from(this.rules.keys()).filter(e => !e.includes("/"))
+  }
+
+  downloadDocs(
+    urlFromPatternId: (patternId: string) => string,
+    prefix: string = ""
+  ) {
+    const patterns =
+      prefix.length > 0
+        ? this.patternIdsWithoutPrefix(prefix)
+        : this.eslintPatternIds()
+    const promises: Promise<void>[] = patterns.map(async pattern => {
+      const url: string = urlFromPatternId(pattern)
+      const result = await fetch(url)
+      if (result.ok) {
+        const text = await result.text()
+        const filename =
+          "docs/description/" +
+          (prefix.length > 0 ? prefix + "_" : "") +
+          patternIdToCodacy(pattern) +
+          ".md"
+        return writeFile(filename, text)
+      } else return Promise.resolve()
```

Comment:
why we are swallowing errors?
`Promise.reject` instead

Reply:
✅Done (found a bug too, thank you 🙏)